module/Application/src/Module.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * @link      http://github.com/zendframework/ZendSkeletonApplication for the canonical source repository
+ * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Application;
+
+use Zend\Mvc\ModuleRouteListener;
+use Zend\Mvc\MvcEvent;
+
+class Module
+{
+    const VERSION = '3.0.0dev';
+
+    public function onBootstrap(MvcEvent $e)
+    {
+        $moduleRouteListener = new ModuleRouteListener();
```

Comment:
I have no clue how often it's used in real-world applications, unfortunately. I suspect it could be useful for having a root route that defines a namespace, but cannot match without child routes, but even then, I personally would rather use FQCN for the controllers anyways.

Definitely up for debate.